crates/re_sdk/src/recording_stream.rs

```diff
@@ -311,6 +315,55 @@ impl RecordingStreamBuilder {
         }
     }
 
+    /// Spawns a new Rerun Viewer process from an executable available in PATH, then creates a new
+    /// [`RecordingStream`] that is pre-configured to stream the data through to that viewer over TCP.
+    ///
+    /// `flush_timeout` is the minimum time the [`TcpSink`][`crate::log_sink::TcpSink`] will
+    /// wait during a flush before potentially dropping data.  Note: Passing `None` here can cause a
+    /// call to `flush` to block indefinitely if a connection cannot be established.
+    ///
+    /// ## Example
+    ///
+    /// ```no_run
+    /// let rec = re_sdk::RecordingStreamBuilder::new("rerun_example_app")
+    ///     .spawn(&re_sdk::SpawnOptions::default(), re_sdk::default_flush_timeout())?;
+    /// # Ok::<(), Box<dyn std::error::Error>>(())
+    /// ```
+    pub fn spawn(
+        self,
+        opts: &crate::SpawnOptions,
+        flush_timeout: Option<std::time::Duration>,
+    ) -> RecordingStreamResult<RecordingStream> {
+        use std::{net::TcpStream, time::Duration};
+
+        let connect_addr = opts.connect_addr();
+
+        // NOTE: If `_RERUN_TEST_FORCE_SAVE` is set, all recording streams will write to disk no matter
+        // what, thus spawning a viewer is pointless (and probably not intended).
+        if forced_sink_path().is_some() {
+            return self.connect(connect_addr, flush_timeout);
+        }
+
+        if TcpStream::connect_timeout(&connect_addr, Duration::from_millis(1)).is_ok() {
+            re_log::info!(
+                addr = %opts.listen_addr(),
+                "A process is already listening at this address. Trying to connect instead."
+            );
+        } else {
+            crate::spawn(opts).map_err(RecordingStreamError::SpawnViewer)?;
+
+            // Give the newly spawned Rerun Viewer some time to bind.
+            for _ in 0..5 {
+                if TcpStream::connect_timeout(&connect_addr, Duration::from_millis(1)).is_ok() {
+                    break;
+                }
+                std::thread::sleep(Duration::from_millis(100));
+            }
```

Comment:
Why not just do a single `connect_timeout`  with 500ms ?

Reply:
If nothing is bound to that address at all yet then `connect` will fail immediately, irrelevant of the timeout configuration.

I'll add a note.